### PR TITLE
Add :protocol parameter to make-channel-socket-client!.

### PR DESCRIPTION
**Pull request by tiensonqin**
Since react native don't support window.location.

### CODE REVIEW

**Comment by danielcompton** [comment]
It's probably better to pass the websocket protocol here ws/wss, rather than http/https

**Comment by tiensonqin** [comment]
@danielcompton Yeah, thanks for advice, I agree with you.

File: src/taoensso/sente.cljc [hunk 1]
```diff
@@ -1310,14 +1310,16 @@
      Common options:
        :type           ; e/o #{:auto :ws :ajax}. You'll usually want the default (:auto).
        :host           ; Server host (defaults to current page's host).
+       :protocol       ; Custom protocol, #{:wss :ws}.
+                       ; React native don't have window.location.
```
Comment (danielcompton) [inline src/taoensso/sente.cljc]: React native doesn't have window.location, so we can't infer the protocol.

File: src/taoensso/sente.cljc [hunk 2]
```diff
@@ -1339,15 +1341,18 @@
      (let [packer (coerce-packer packer)
 
            win-loc (enc/get-win-loc)
-           win-protocol  (:protocol win-loc)
+           protocol (case (keyword protocol)
+                      :wss "https:"
+                      :ws  "http:"
+                      (:protocol win-loc))
```
Comment (danielcompton) [inline src/taoensso/sente.cljc]: If we're going to allow the user to specify a protocol, then that should go first. Probably an `or` here would be appropriate?

File: src/taoensso/sente.cljc [hunk 3]
```diff
@@ -1339,15 +1341,18 @@
      (let [packer (coerce-packer packer)
 
            win-loc (enc/get-win-loc)
-           win-protocol  (:protocol win-loc)
+           protocol (case (keyword protocol)
```
Comment (danielcompton) [inline src/taoensso/sente.cljc]: We need to get the protocol from `win-loc` if it's not provided.

Comment (danielcompton) [inline src/taoensso/sente.cljc]: Probably good to add an assert if `win-loc` is nil and `protocol` not provided, and in the error explain that `protocol` must be provided.

**Comment by danielcompton** [comment]
Hmm, don't have time to look at it this second, but how does the Node client do this? It too won't have a `window.location`

**Comment by ptaoussanis** [comment]
Merged, thanks!

**Pull request closed without merging**